Adapt web socket

juljac56 opened the pull request:
Worked on file StudentMeeting.tsx, in the folder Student

Changed made because the websocket communication structure changed in the back (Fast API) : 

- Add an ID to the url of the websocket, so that each student has its own webscoket to communicate with the back
- Add the roomName in the url of the websocket
- Creation of the websocket on launch of the front or on change of the roomName.

See branch websocket_protoco, to understand the change made in the back

juljac56 committed:
change websocket in useEffect

juljac56 committed:
Merge branch 'main' into adapt_WebSocket

juljac56 committed:
🐛(studentMeeting.tsx) change name of parameters in url request

change the name of parameters because the names og the variables changed in the back

juljac56 committed:
🔥(studentMeeting.tsx) remove function interval to send websocket

remove useless send of webscocket message from the front

AladdinVR commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -65,6 +65,9 @@ const StudentMeeting: FunctionComponent = () => {
     const [loading, setLoading] = useState<boolean>(false);
 
+    //websocket
+    const [ws1, setWs1] = useState<WebSocket>();
```
```suggestion
    const [ws, setWs] = useState<WebSocket>();
```
we use only one socket

AladdinVR commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -81,29 +84,31 @@ const StudentMeeting: FunctionComponent = () => {
-        if (addressWebsocket == undefined) {
+        const id = Date.now();
+        const addressWebsocket_general = process.env.REACT_APP_WS_ADDRESS;
+        if (addressWebsocket_general == undefined) {
```
```suggestion
        if (addressWebsocket_general === undefined) {
```

juljac56 committed:
🎨(StudentMeeting.tsx) take into account reviews about name of variable

change the name of the variable used to define the websocket

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -81,29 +84,31 @@ const StudentMeeting: FunctionComponent = () => {
     useEffect(() => {
-        const addressWebsocket = process.env.REACT_APP_WS_ADDRESS;
-        if (addressWebsocket == undefined) {
+        const id = Date.now();
```
It is not a good idea to use the date creation as an identifier even if those are probably unique it is better something made for that. You could either generate it server-side and send it back to the client or use uuid (faster to implement here) which are made for this kind of usage.

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -81,29 +84,31 @@ const StudentMeeting: FunctionComponent = () => {
             };
         }
-    }, [processSelected, information]);
+    }, [ws, processSelected, information]);
```
You don't information in the dependency here, you don't use it in the callback.

juljac56 committed:
➕(StudentMeeting) add dependency uuid for the websocket

add dependency uuid to generate an id for the creation of the webscoket for each client, the id used to be generated with date.now()
